src/cz/cvut/cognitive/override/DistractionScreenGUIController.java

```diff
@@ -175,323 +121,113 @@ public void onStartScreen()
         {   
-                DistractionSettings.setProbabilityPedestrian(Slider_pedestrian.getValue());
                 Value_pedestrian.clear();
                 Value_pedestrian.addItem(Slider_pedestrian.getValue());
         }
         
         @NiftyEventSubscriber(id="Slider_text") 
         public void propabilityText(final String id, final SliderChangedEvent event) 
         {   
-                DistractionSettings.setProbabilityText(Slider_text.getValue());
                 Value_text.clear();
                 Value_text.addItem(Slider_text.getValue());
         }
         
         @NiftyEventSubscriber(id="Slider_dark") 
         public void propabilityDark(final String id, final SliderChangedEvent event) 
         {   
-                DistractionSettings.setProbabilityDark(Slider_dark.getValue());
                 Value_dark.clear();
                 Value_dark.addItem(Slider_dark.getValue());
         }
         
         @NiftyEventSubscriber(id="Slider_collect") 
         public void propabilityCollect(final String id, final SliderChangedEvent event) 
         {   
-                DistractionSettings.setProbabilityCollect(Slider_collect.getValue());
                 Value_collect.clear();
                 Value_collect.addItem(Slider_collect.getValue());
         }
         
         
+        @Override
 	public void clickStartButton()
 	{
-                DistractionSettings.setDistScenario(true);
-		DistractionClass.initialize(sim);
-                instructionScreenGUI.hideDialog();
+		// Initialize all distractors here: 
+                new BoxDistraction(sim, 5f, Slider_box.getValue(), 2f, 1, 1, 1, 2, "Textures"+File.separator+"DistractionTask"+File.separator+"default_box.jpg");
+                new SoundDistraction(sim, 2f, Slider_sound.getValue(), 1f);
+                new DarkeningDistraction(sim, 1f, Slider_dark.getValue(), 3f);
+                new PedestrianDistraction(sim, 30f, Slider_pedestrian.getValue(), 5f, "Textures"+File.separator+"DistractionTask"+File.separator+"default_pedestrian.jpg");
+                new TextDistraction(sim, 4f, Slider_text.getValue(), 3f);
+                new CollectObjectDistraction(sim, 5f, Slider_collect.getValue(), 3f, "Textures"+File.separator+"DistractionTask"+File.separator+"default_greenSphere.png", "Textures"+File.separator+"DistractionTask"+File.separator+"default_redSphere.png");
+                new WeatherDistraction(sim, 1f, Slider_rain.getValue(), WeatherDistraction.Type.RAIN); 
+                new WeatherDistraction(sim, 1f, Slider_snow.getValue(), WeatherDistraction.Type.SNOW); 
+                new WeatherDistraction(sim, 1f, Slider_fog.getValue(), WeatherDistraction.Type.FOG); 
+                
+                super.clickStartButton();
 	}
         
-          public void clickDefButton()
+        @NiftyEventSubscriber(id="DefaultButton")
+          private void clickDefButton() //FIXME I broke the default button - GUI is not reset
```

Comment:
@JohnnyMarek I broke the DefaultButton, even though it should work. Also the values are shown initially only for some. Can you take a look please?